stdlib/public/core/StringStorage.swift

```diff
@@ -726,6 +774,48 @@ final internal class _SharedStringStorage: _AbstractStringStorage {
     
     return nil
   }
+  
+  @objc(UTF8String)
+  @_effects(readonly)
+  final internal func _utf8String() -> UnsafePointer<UInt8>? {
+    return start
+  }
+
+  @objc(cStringUsingEncoding:)
+  @_effects(readonly)
+  final internal func _cString(encoding: UInt) -> UnsafePointer<UInt8>? {
+    switch (encoding, isASCII) {
+    case (_cocoaASCIIEncoding, true):
+      fallthrough
+    case (_cocoaUTF8Encoding, _):
+      return start
+    default:
+      return _cocoaCStringUsingEncodingTrampoline(self, encoding)
+    }
+  }
+  
+  @objc(getCString:maxLength:encoding:)
+  @_effects(releasenone)
+  final internal func getCString(_ outputPtr: UnsafeMutablePointer<UInt8>,
+                               maxLength: Int,
+                               encoding: UInt)
+    -> Int8 {
+    switch (encoding, isASCII) {
+    case (_cocoaASCIIEncoding, true):
+      fallthrough
+    case (_cocoaUTF8Encoding, _):
+      guard maxLength + 1 >= count else { return 0 }
+      let buffer = UnsafeMutableBufferPointer(start: outputPtr, count: maxLength)
+      let (_, end) = asString.utf8._copyContents(initializing: buffer)
+      buffer[end] = 0
+      return 1
+    default:
+      return  _cocoaGetCStringTrampoline(self,
+                                         outputPtr,
+                                         maxLength,
+                                         encoding)
+    }
+  }
```

Comment:
Rather than duplicate the bodies, can we have a helper function that just takes `start` and does this? Can that be the equivalent of `_cocoaGetCStringTrampoline`?